lib/improver/tests/weighted_blend/test_TriangularWeightedBlendAcrossAdjacentPoints.py

```diff
@@ -0,0 +1,227 @@
+                                                  input_new_cube,
+                                                  ['longitude'])
+        self.assertEqual(input_new_cube.coord('longitude'),
+                         self.orig_cube.coord('longitude'))
+        self.assertEqual(input_new_cube.coord('latitude'),
+                         self.new_cube.coord('latitude'))
+        self.assertEqual(input_new_cube.coord('forecast_period'),
+                         self.new_cube.coord('forecast_period'))
+
+    def test_change_two_coord(self):
+        """Test it corrects multiple coordinate"""
+        input_new_cube = self.new_cube.copy()
+        self.plugin.correct_collapsed_coordinates(
+            self.orig_cube, input_new_cube, ['longitude', 'forecast_period'])
+        self.assertEqual(input_new_cube.coord('longitude'),
+                         self.orig_cube.coord('longitude'))
+        self.assertEqual(input_new_cube.coord('latitude'),
+                         self.new_cube.coord('latitude'))
+        self.assertEqual(input_new_cube.coord('forecast_period'),
+                         self.orig_cube.coord('forecast_period'))
+
+    def test_bounds_corrected(self):
+        """Test it corrects bounds"""
+        input_new_cube = self.new_cube.copy()
+        input_orig_cube = self.orig_cube.copy()
+        input_orig_cube.remove_coord('forecast_period')
+        input_orig_cube.add_aux_coord(DimCoord(
+            [5, 6], "forecast_period", bounds=[[4.5, 5.5], [5.5, 6.5]],
+            units="hours"), 0)
+        self.plugin.correct_collapsed_coordinates(
+            input_orig_cube, input_new_cube, ['forecast_period'])
+        self.assertEqual(input_new_cube.coord('forecast_period'),
+                         input_orig_cube.coord('forecast_period'))
+
+    def test_wrong_size_coords(self):
+        """Test it raises an error when it new_cube and old_cube have
+           different length coordinates"""
+        data = np.zeros((2, 2, 2))
+        orig_cube = Cube(data, units="m",
+                         standard_name="lwe_thickness_of_precipitation_amount")
+        orig_cube.add_dim_coord(DimCoord([0, 1], "forecast_period",
+                                         units="hours"), 0)
+        data = np.zeros((3, 2, 2))
+        new_cube = Cube(data, units="m",
+                        standard_name="lwe_thickness_of_precipitation_amount")
+        new_cube.add_dim_coord(DimCoord([0, 1, 2], "forecast_period",
+                                        units="hours"), 0)
+        message = "New points shape must match existing points shape."
+        with self.assertRaisesRegexp(ValueError, message):
+            self.plugin.correct_collapsed_coordinates(orig_cube, new_cube,
+                                                      ['forecast_period'])
+
+
+class Test_process(IrisTest):
+
+    """Test the process method."""
+
+    def setUp(self):
+        """Set up a test cube."""
+        data = np.zeros((2, 2, 2))
```

Comment:
I don't think this cube set up is very different to existing `set_up_cube` functions.